src/Symfony/Component/Console/Tests/DependencyInjection/AddConsoleCommandPassTest.php

```diff
@@ -76,6 +76,23 @@ public function testProcessRegisterLazyCommands()
         $this->assertSame(array('console.command.symfony_component_console_tests_dependencyinjection_mycommand' => false), $container->getParameter('console.command.ids'));
     }
 
+    /**
+     * @expectedException        \Symfony\Component\DependencyInjection\Exception\InvalidArgumentException
+     * @expectedExceptionMessage  Service "my-command" has multiple "console.command" tags with a different value for the "command" attribute, but the command must have only one name. Use the "alias" attribute instead.
+     */
+    public function testProcessThrowsIfCommandHasMultipleNames()
+    {
+        $container = new ContainerBuilder();
+        $container
+            ->register('my-command', MyCommand::class)
+            ->setPublic(false)
+            ->addTag('console.command', array('command' => 'my:command'))
+            ->addTag('console.command', array('command' => 'other:name'))
+        ;
```

Comment:
Still think this is the best option actually :)

> specially because relying on the order on which tags are defined sounds less good to me

I guess it aims for to be explicit what getName returns, but do we care? It's the first one :) and practically there's no difference either. Low level perhaps, but thats not an issue for end users.

Alternatively; `command: name, aliases: [name, nameN]` and dont allow `console.command` more than once.

Alter-alternatively; perhaps `command` is a bad name due term duplication with `console.command`, what about only `name`?